add instructions to train FLAVA model (#347)
facebookresearch/multimodal: examples/flava/native/README.md

# Usage Instructions

> 

Reviewer: Can we add some description here i.e. this is a native pytorch lightweight implementation to run scaling studies which is subset of original code. For original code , people should follow the other script in the FLAVA directory

## Prerequisites

> - Installed PyTorch 1.13 or nightly 

Reviewer 2: ```suggestion
- Install PyTorch 1.13 or nightly 
```

> - `git clone https://github.com/facebookresearch/multimodal`

Reviewer: just link to the library readme as first step and pip install on the flava requirements as second step? that should be enough?

Author: the examples/ folder isn't included if users install the library via pip:

https://github.com/facebookresearch/multimodal/blob/main/setup.py#L80

Reviewer: you need to link to this https://github.com/facebookresearch/multimodal/blob/main/README.md#building-from-source

# Usage Instructions

This is a lightweight native pytorch implementation to run scaling studies on the FLAVA model. The original code is located at: [`examples/flava/train.py`](https://github.com/facebookresearch/multimodal/blob/main/examples/flava/train.py)

## Prerequisites

> - Install PyTorch 1.13 or nightly 

Reviewer: you dont need this line?

> - Install [torchmultimodal library](https://github.com/facebookresearch/multimodal#installation)

Reviewer: not needed? lib readme will cover

- `git clone https://github.com/facebookresearch/multimodal`

> - `cd multimodal/examples`

Reviewer: still need to do pip install of flava requirements to get HF dependencies

## Training 

### Configuration

Configuration presets can be found at: `examples/flava/native/configs`

> Some config settings that are relevant for scaling: `batch_size`, `activation_checkpointing`, `strategy`.

Reviewer: ```suggestion
Some config settings that are relevant for scaling: (local)`batch_size`, `activation_checkpointing`, `strategy`.
```

> Configs can be overridden through command line, for example: `python -m flava.native.train config=flava/native/configs/2.7b.yaml training.batch_size=8 training.enable_amp=True training.activation_checkpointing=True training.strategy=fsdp`

Reviewer: ```suggestion
Configs can be overridden through command line, for example: `python -m flava.native.train config=flava/native/configs/pretrain_debug.yaml training.batch_size=8 training.enable_amp=True training.activation_checkpointing=True training.strategy=fsdp`
```
